lib/node_modules/@stdlib/array/complex64/lib/main.js

```diff
@@ -1522,6 +1523,49 @@ setReadOnly( Complex64Array.prototype, 'indexOf', function indexOf( searchElemen
 	return -1;
 });
 
+/**
+* Returns a new string by concatenating all array elements.
+*
+* @name join
+* @memberof Complex64Array.prototype
+* @type {Function}
+* @param {string} [separator=','] - element separator
+* @throws {TypeError} `this` must be a complex number array
+* @throws {TypeError} first argument must be a string
+* @returns {string} string representation
+*
+* @example
+* var arr = new Complex64Array( 2 );
+*
+* arr.set( [ 1.0, 1.0 ], 0 );
+* arr.set( [ 2.0, 2.0 ], 1 );
+*
+* var str = arr.join();
+* // returns '1 + 1i,2 + 2i'
+*
+* str = arr.join( '/' );
+* // returns '1 + 1i/2 + 2i'
+*/
+setReadOnly( Complex64Array.prototype, 'join', function join( separator ) {
+	var out;
+	var buf;
+	var i;
+	if ( !isComplexArray( this ) ) {
+		throw new TypeError( 'invalid invocation. `this` is not a complex number array.' );
+	}
+	if (arguments.length === 0) {
+		separator = ',';
+	} else if ( typeof separator !== 'string' ) {
```

Comment:
```suggestion
	} else if ( !isString( separator ) ) {
```

Use the dedicated assertion package.